### DISCUSSION TITLE
Full-width space inserted instead of language switch on pressing Super+Space in Hiragana Input mode in Fedora 39

### DISCUSSION THREAD

**Discussion by earlydogs** [google/mozc#854] [Q&A] [answered]
## Description
In Fedora 39, when the input mode is set to Hiragana and the language switch key (Super+Space) is pressed, it fails to switch languages and instead inserts a full-width space. This issue occurs during input in Terminal applications but does not happen in browser applications like Chrome. Initially thought to be a Fedora 39 bug, this issue does not occur with other input sources (such as Anthy), leading to this report.

## Steps to reproduce
1. Set up Fedora 39.
2. Install ibus-mozc with sudo dnf install ibus-mozc.
3. Restart the operating system.
4. Add Japanese (Mozc) to Input sources through the Settings menu.
5. Switch to Mozc by pressing Super + Space.
6. Change the Mozc setting to Hiragana in Input Mode.
7. In a terminal application, press Super + Space.

## Version or commit-id
Installed Packages
Name         : ibus-mozc
Version      : 2.29.5111.102
Release      : 3.fc39
Architecture : x86_64
Size         : 1.1 M
Source       : mozc-2.29.5111.102-3.fc39.src.rpm
Repository   : @System
From repo    : fedora
Summary      : The mozc engine for IBus input platform
URL          : https://github.com/google/mozc
License      : BSD-3-Clause AND Apache-2.0 AND Unicode-DFS-2015 AND NAIST-2003
Description  : Mozc is a Japanese Input Method Editor (IME) designed for
             : multi-platform such as Chromium OS, Windows, Mac and Linux.
             : 
             : This package contains the Input Method Engine for IBus.

Installed Packages
Name         : mozc
Version      : 2.29.5111.102
Release      : 3.fc39
Architecture : x86_64
Size         : 23 M
Source       : mozc-2.29.5111.102-3.fc39.src.rpm
Repository   : @System
From repo    : fedora
Summary      : A Japanese Input Method Editor (IME) designed for multi-platform
URL          : https://github.com/google/mozc
License      : BSD-3-Clause AND Apache-2.0 AND Unicode-DFS-2015 AND NAIST-2003
Description  : Mozc is a Japanese Input Method Editor (IME) designed for
             : multi-platform such as Chromium OS, Windows, Mac and Linux.

## Environment
 - OS: Fedora 39

**Comment by hiroyuki-komatsu** [comment]
Would you also check other modifier keys rather than `Super` to switch the input source (e.g. Shift + Space) ?

**Reply by earlydogs** [reply]
Yes. Even when set to Shift+Space instead of Super+Space, a full-width space is entered.

**Comment by earlydogs** [comment, marked as answer]
The issue was caused by Wayland. Disabling Wayland resolved the issue. Please close this thread. Thank you.

**Comment by hiroyuki-komatsu** [comment]
Thank you for the update.
I marked your comment as the answer of this Q&A.

**Comment by phoepsilonix** [comment]
PR #1059 resolves this issue. I tested it on Manjaro Linux(GNOME/Wayland) and confirmed that it works well.

**Comment by jdkoren** [comment]
I'm on PikaOS (debian-based) with `Mozc-2.29.5160.102`, GNOME, and Wayland, and this issue is reproducible.

**Reply by hiroyuki-komatsu** [reply]
2.30.5618 or later version includes this fix.

Note, 2.30.5618 is the version on 2024-10-10, and 2.29.5160 is the version on 2023-07-28.

https://github.com/google/mozc/tags

**Reply by jdkoren** [reply]
I'm pretty sure I used `sudo apt install ibus-mozc`, which resulted in the version i have now. Is that the correct way to install on Debian-based systems, or have I done something wrong?

**Reply by hiroyuki-komatsu** [reply]
I am not the maintainer of the Debian package, but version 2.29.5160 appears to be the latest one available.

Therefore, you might want to manually build and install the version that includes this fix.

https://github.com/google/mozc/blob/master/docs/build_mozc_for_linux.md

**Reply by yukawa** [reply]
> Is that the correct way to install on Debian-based systems, or have I done something wrong?

According to https://wiki.pika-os.com/en/faqs, the recommended way to ask for help is to reach out their discord channel, I guess.

 * https://wiki.pika-os.com/en/faqs#h-2-if-not-ask-for-help-on-our-discord-forum

> ### I have encountered a problem on PikaOS, what should I do?
> #### 2 - If not, ask for help on our Discord forum
> [#help forum channel](https://canary.discord.com/channels/1077709872314982450/1077737208687108216)
